ledger/error/src/main/scala/com/daml/error/ErrorResource.scala

```diff
@@ -48,6 +48,9 @@ object ErrorResource {
   object ContractKey extends ErrorResource {
     def asString: String = "CONTRACT_KEY"
   }
+  object SharedKey extends ErrorResource {
+    def asString: String = "SHARED_KEY"
+  }
```

Comment:
I think I am fine we call the new concept a shared key internally, 
but I do not think we want to expose this concept to the Daml user. 
Long run, we just change the way contract hash is calculating.

Reply:
When an error is parsed by the `GrpcErrorParser` a `GlobalKey` is often needed. To reconstruct a GlobalKey In addition to templateId and value one of the following is needed:

1. - A flag to indicate if the 
2. - A way to resolve templateId to transaction version
3. - The GlobalKey hash

I have currently gone for option #2 as that is the most pragmatic and this info is only leaked in an error situation.

Reply:
If you manage to implement option 2, is the way to go.

Reply:
Following an investigation left as option 1 for now, will do follow up PR as necessary.